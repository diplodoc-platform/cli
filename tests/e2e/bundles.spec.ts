import {describe, it} from 'vitest';
import {TestAdapter, compareDirectories, getTestPaths} from '../fixtures';

describe('Check bundles', () => {
    it('bundles list is correct', async () => {
        const {inputPath, outputPath} = getTestPaths('mocks/bundles');

        await TestAdapter.testBuildPass(inputPath, outputPath, {
            md2md: false,
            md2html: true,
            args: '-j2',
        });
        await compareDirectories(outputPath, true, true);
    });
});
