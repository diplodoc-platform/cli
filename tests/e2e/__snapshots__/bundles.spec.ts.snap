// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Check bundles > bundles list is correct > filelist 1`] = `
"[
  ".yfm",
  "_bundle/app-css-1",
  "_bundle/app-js-2",
  "_bundle/latex-extension.css",
  "_bundle/latex-extension.js",
  "_bundle/mermaid-extension.js",
  "_bundle/page-constructor-extension.css",
  "_bundle/page-constructor-extension.js",
  "_bundle/search-async-0",
  "_bundle/search-async-1",
  "_bundle/search-css-0",
  "_bundle/search-css-1",
  "_bundle/search-css-2",
  "_bundle/search-css-3",
  "_bundle/search-css-4",
  "_bundle/search-js-0",
  "_bundle/search-js-1",
  "_bundle/search-js-2",
  "_search/api.js",
  "_search/ru/hash-index.js",
  "_search/ru/hash-registry.js",
  "_search/ru/hash-resources.js",
  "_search/ru/index.html",
  "_search/ru/language.js",
  "index.html",
  "page1.html",
  "page2.html",
  "toc.js"
]"
`;

exports[`Check bundles > bundles list is correct 1`] = `
"allowHTML: true

meta:
  rootPath: http://127.0.0.1:5000/

interface:
  toc-header: false
  favicon-src: https://storage.yandexcloud.net/diplodoc-www-assets/favicon/favicon.ico

search:
  provider: local
  tolerance: 2
  confidense: phrased
"
`;

exports[`Check bundles > bundles list is correct 2`] = `
"<!DOCTYPE html>
<html lang="ru" dir="ltr">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <base href="./" />
        <title>Header | Skip html extension</title>
        <meta name="generator" content="Diplodoc Platform vDIPLODOC-VERSION">
        <meta name="availableLangs" content="en,ru">
        <meta http-equiv="Content-Security-Policy" content="worker-src 'self';">
        <style type="text/css">html, body {min-height:100vh; height:100vh;}</style>
        <link rel="icon" type="image/x-icon" href="https://storage.yandexcloud.net/diplodoc-www-assets/favicon/favicon.ico">
        
        <link type="text/css" rel="stylesheet" href="_bundle/search-css-0"/>
        <link type="text/css" rel="stylesheet" href="_bundle/app-css-1"/>
    </head>
    <body class="g-root g-root_theme_light">
        <div id="root"></div>
        <script type="application/json" id="diplodoc-state">
            {"data":{"leading":false,"html":"&lt;p&gt;Content&lt;/p&gt;/n&lt;p&gt;&lt;a href=\\"page1.html\\"&gt;Link 1&lt;/a&gt;&lt;br /&gt;/n&lt;a href=\\"page2.html\\"&gt;Link 2&lt;/a&gt;&lt;/p&gt;/n","meta":{"metadata":[{"name":"generator","content":"Diplodoc Platform vDIPLODOC-VERSION"}],"availableLangs":["en","ru"]},"headings":[],"title":"Header"},"router":{"pathname":"index","depth":1,"base":"./"},"lang":"ru","langs":["ru"],"viewerInterface":{"toc-header":false,"favicon-src":"https://storage.yandexcloud.net/diplodoc-www-assets/favicon/favicon.ico","toc":true,"search":true,"feedback":true},"search":{"provider":"local","tolerance":2,"confidense":"phrased","enabled":true,"api":"_search/api.js","link":"_search/ru/index.html"}}
        </script>
        <script type="application/javascript">
            const data = document.querySelector('script#diplodoc-state');
            window.__DATA__ = JSON.parse((function unescape(string) {
                return string.replace(/&lt;/g, "<").replace(/&gt;/g, ">").replace(/&amp;/g, "&");
              })(data.innerText));
            window.STATIC_CONTENT = false;
        </script>
        <script type="application/javascript" defer src="toc.js"></script>
        <script type="application/javascript" defer src="_search/ru/hash-resources.js"></script>
        <script type="application/javascript" defer src="_bundle/search-js-1"></script>
        <script type="application/javascript" defer src="_bundle/search-js-2"></script>
        <script type="application/javascript" defer src="_bundle/app-js-2"></script>
        
    </body>
</html>"
`;

exports[`Check bundles > bundles list is correct 3`] = `
"<!DOCTYPE html>
<html lang="ru" dir="ltr">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <base href="./" />
        <title>Page 1 | Skip html extension</title>
        <meta name="generator" content="Diplodoc Platform vDIPLODOC-VERSION">
        <meta http-equiv="Content-Security-Policy" content="worker-src 'self';">
        <style type="text/css">html, body {min-height:100vh; height:100vh;}</style>
        <link rel="icon" type="image/x-icon" href="https://storage.yandexcloud.net/diplodoc-www-assets/favicon/favicon.ico">
        
        <link type="text/css" rel="stylesheet" href="_bundle/search-css-0"/>
        <link type="text/css" rel="stylesheet" href="_bundle/app-css-1"/>
    </head>
    <body class="g-root g-root_theme_light">
        <div id="root"></div>
        <script type="application/json" id="diplodoc-state">
            {"data":{"leading":false,"html":"&lt;p&gt;&lt;a href=\\"page2.html#hash\\"&gt;Link&lt;/a&gt;&lt;/p&gt;/n","meta":{"metadata":[{"name":"generator","content":"Diplodoc Platform vDIPLODOC-VERSION"}]},"headings":[],"title":"Page 1"},"router":{"pathname":"page1","depth":1,"base":"./"},"lang":"ru","langs":["ru"],"viewerInterface":{"toc-header":false,"favicon-src":"https://storage.yandexcloud.net/diplodoc-www-assets/favicon/favicon.ico","toc":true,"search":true,"feedback":true},"search":{"provider":"local","tolerance":2,"confidense":"phrased","enabled":true,"api":"_search/api.js","link":"_search/ru/index.html"}}
        </script>
        <script type="application/javascript">
            const data = document.querySelector('script#diplodoc-state');
            window.__DATA__ = JSON.parse((function unescape(string) {
                return string.replace(/&lt;/g, "<").replace(/&gt;/g, ">").replace(/&amp;/g, "&");
              })(data.innerText));
            window.STATIC_CONTENT = false;
        </script>
        <script type="application/javascript" defer src="toc.js"></script>
        <script type="application/javascript" defer src="_search/ru/hash-resources.js"></script>
        <script type="application/javascript" defer src="_bundle/search-js-1"></script>
        <script type="application/javascript" defer src="_bundle/search-js-2"></script>
        <script type="application/javascript" defer src="_bundle/app-js-2"></script>
        
    </body>
</html>"
`;

exports[`Check bundles > bundles list is correct 4`] = `
"<!DOCTYPE html>
<html lang="ru" dir="ltr">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <base href="./" />
        <title>Page 2 | Skip html extension</title>
        <meta name="generator" content="Diplodoc Platform vDIPLODOC-VERSION">
        <meta http-equiv="Content-Security-Policy" content="worker-src 'self';">
        <style type="text/css">html, body {min-height:100vh; height:100vh;}</style>
        <link rel="icon" type="image/x-icon" href="https://storage.yandexcloud.net/diplodoc-www-assets/favicon/favicon.ico">
        
        <link type="text/css" rel="stylesheet" href="_bundle/search-css-0"/>
        <link type="text/css" rel="stylesheet" href="_bundle/app-css-1"/>
    </head>
    <body class="g-root g-root_theme_light">
        <div id="root"></div>
        <script type="application/json" id="diplodoc-state">
            {"data":{"leading":false,"html":"&lt;h2 id=\\"hash\\"&gt;&lt;a href=\\"page2.html#hash\\" class=\\"yfm-anchor\\" aria-hidden=\\"true\\"&gt;&lt;span class=\\"visually-hidden\\" data-no-index=\\"true\\"&gt;hash&lt;/span&gt;&lt;/a&gt;hash&lt;/h2&gt;/n&lt;p&gt;&lt;a href=\\"https://example.com\\" target=\\"_blank\\" rel=\\"noreferrer noopener\\"&gt;External link&lt;/a&gt;&lt;/p&gt;/n","meta":{"metadata":[{"name":"generator","content":"Diplodoc Platform vDIPLODOC-VERSION"}]},"headings":[{"title":"hash","href":"page2.html#hash","level":2}],"title":"Page 2"},"router":{"pathname":"page2","depth":1,"base":"./"},"lang":"ru","langs":["ru"],"viewerInterface":{"toc-header":false,"favicon-src":"https://storage.yandexcloud.net/diplodoc-www-assets/favicon/favicon.ico","toc":true,"search":true,"feedback":true},"search":{"provider":"local","tolerance":2,"confidense":"phrased","enabled":true,"api":"_search/api.js","link":"_search/ru/index.html"}}
        </script>
        <script type="application/javascript">
            const data = document.querySelector('script#diplodoc-state');
            window.__DATA__ = JSON.parse((function unescape(string) {
                return string.replace(/&lt;/g, "<").replace(/&gt;/g, ">").replace(/&amp;/g, "&");
              })(data.innerText));
            window.STATIC_CONTENT = false;
        </script>
        <script type="application/javascript" defer src="toc.js"></script>
        <script type="application/javascript" defer src="_search/ru/hash-resources.js"></script>
        <script type="application/javascript" defer src="_bundle/search-js-1"></script>
        <script type="application/javascript" defer src="_bundle/search-js-2"></script>
        <script type="application/javascript" defer src="_bundle/app-js-2"></script>
        
    </body>
</html>"
`;

exports[`Check bundles > bundles list is correct 5`] = `"window.__DATA__.data.toc = {"title":"Skip html extension","href":"index.html","items":[{"name":"Title 1","href":"page1.html","id":"UUID"},{"name":"Title 2","href":"page2.html","id":"UUID"}],"path":"toc.yaml","id":"UUID"};"`;
